jf_agent/jf_jira/jira_download.py

```diff
@@ -1,14 +1,178 @@
+
+    if ingest_config.jira_config.feature_flags.get(
+        MAKARA_CUSTOM_FIELD_MISMATCH_AND_DETECTION_FLAG, False
+    ):
+        try:
+            ids_to_redownload = detect_and_repair_custom_fields(ingest_config=ingest_config)
+            if ids_to_redownload:
+                count_of_ids_to_redownload_previously = len(
+                    ingest_config.jira_config.jellyfish_issue_ids_for_redownload
+                )
+                ingest_config.jira_config.jellyfish_issue_ids_for_redownload.update(
+                    ids_to_redownload
+                )
+                count_of_additional_ids_to_redownload = (
+                    len(ingest_config.jira_config.jellyfish_issue_ids_for_redownload)
+                    - count_of_ids_to_redownload_previously
+                )
+                logging_helper.send_to_agent_log_file(
+                    f'Detect and Repair Custom Fields found {len(ids_to_redownload)} to redownload, '
+                    f'which gave us an additional {count_of_additional_ids_to_redownload} unique IDs to redownload'
+                )
+            else:
+                logger.info('No Jira Issues were marked as needing their custom fields repaired')
+        except Exception as e:
+            logger.warning(
+                f'Exception {e} encountered when attempting to run {detect_and_repair_custom_fields.__name__}.'
+            )
+            logging_helper.send_to_agent_log_file(traceback.format_exc())
+    else:
+        logging_helper.send_to_agent_log_file(
+            f'{MAKARA_CUSTOM_FIELD_MISMATCH_AND_DETECTION_FLAG} was set to {ingest_config.jira_config.feature_flags.get(MAKARA_CUSTOM_FIELD_MISMATCH_AND_DETECTION_FLAG, False)} - Detect and repair custom field logic will not be run',
+            level=logging.INFO,
+        )
+    try:
+        logger.info(f'Attempting to use JF Ingest for Jira Ingestion')
+        success = load_and_push_jira_to_s3(ingest_config)
+        download_status = 'success' if success else 'failed'
+        logger.info(
+            'Jira Download Complete',
+        )
+    except Exception as e:
+        download_status = 'failed'
+        logger.error(
+            'Error encountered when downloading Jira data. '
+            'This Jira submission will be marked as failed. '
+            f'Error: {e}'
+        )
+        logging_helper.send_to_agent_log_file(traceback.format_exc(), level=logging.ERROR)
+    finally:
+        return {'type': 'Jira', 'status': download_status}
+
+
+@diagnostics.capture_timing()
+@logging_helper.log_entry_exit(logger)
+def detect_and_repair_custom_fields(
+    ingest_config: IngestionConfig, submit_issues_for_repair: Optional[bool] = True
+) -> set[str]:
+
+    if not ingest_config.jira_config.feature_flags[MAKARA_CUSTOM_FIELD_MISMATCH_AND_DETECTION_FLAG]:
+        logger.info('Skipping detection and repair of custom fields, feature flag is disabled')
+        return set()
```

Comment:
Is this required since we're already checking if the feature flag is enabled in `run_jira_download`?

Additionally, is there any chance that the key `MAKARA_CUSTOM_FIELD_MISMATCH_AND_DETECTION_FLAG` may not exist in `ingest_config.jira_config.feature_flags` (thus causing an exception)? Might be safer to use `.get()` here, but I'm not super familiar with how we interact with FF's

Reply:
Definitely better to use `get()` here, with the `default` value being `False`. Good call on this being redundant, I'm just gonna get rid of it